compiler/src/iree/compiler/Dialect/HAL/Target/VulkanSPIRV/VulkanSPIRVTarget.cpp

```diff
@@ -141,6 +142,13 @@ class VulkanSPIRVTargetBackend : public TargetBackend {
              << "should only contain exactly one spirv.module op";
     }
     auto spvModuleOp = *spirvModuleOps.begin();
+    if (!options.dumpIntermediatesPath.empty()) {
```

Comment:
How do you feel about also disassembling and outputting .spvasm? The WebGPU target does that already: https://github.com/openxla/iree/blob/2dddf02a12a2219fd80d457b5bf32dd155eb3b66/compiler/src/iree/compiler/Dialect/HAL/Target/WebGPU/WebGPUTarget.cpp#L176-L195

That requires a dep on spirv-tools, which WebGPU uses via Tint already... we don't include spirv-tools elsewhere right now. I think _if_ spirv-tools is already included (e.g. WebGPU is enabled), we might as well use it here too.

Reply:
(Or can we get spvasm from `spirv::serialize` or similar, without going through spirv-tools?)

Reply:
FWIW, I actually find the spirv dialect more readable than the output of spirv-dis 😛

Reply:
Well, actually it doesn't hurt to dump whatever we can dump: `spirv.module`, `.spv`, or `.spvasm`. :)